OpenTokMeet/app/src/main/java/com/opentok/android/meet/HomeActivity.java

```diff
@@ -202,23 +203,39 @@ public void onClick(DialogInterface dialog, int id) {
                 }
             }
         };
+        try {
         /* update values from ui */
-        mRoomName   = ((EditText)findViewById(R.id.input_room_name)).getText().toString();
-        mUsername   = ((EditText)findViewById(R.id.input_username)).getText().toString();
-        mH264Support= ((Switch)findViewById(R.id.h264Support)).isChecked();
+            mRoomName = ((EditText) findViewById(R.id.input_room_name)).getText().toString();
+            mUsername = ((EditText) findViewById(R.id.input_username)).getText().toString();
+            mH264Support = ((Switch) findViewById(R.id.h264Support)).isChecked();
         /* save conference settings */
-        saveSettings();
+            saveSettings();
         /* set debug settings (setup from advanced settings menu) */
-        updatePreferences();
+            updatePreferences();
+
+            //Replace all leading spaces
+            mRoomName = mRoomName.replaceAll("^\\s+", "");
         /* make sure there is a room name defined */
-        if (!mRoomName.isEmpty()) {
+            mRoomName = encodeSpecialCharacters(mRoomName);
+
+
+            if (!mRoomName.isEmpty()) {
             /* request conference information from server */
-            fetchRoomData.execute(mRoomName);
-        } else {
```

Comment:
If the room name is empty but it does not fail encoding, there will be no error to the user... maybe have it throw an exception or put an if block here too with a different toast message?